contracts/modules/Checkpoint/Automation/ScheduleCheckpoint.sol

```diff
@@ -0,0 +1,269 @@
+pragma solidity 0.5.8;
+
+import "../ICheckpoint.sol";
+import "../../TransferManager/TransferManager.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "../../../libraries/BokkyPooBahsDateTimeLibrary.sol";
+import "./ScheduleCheckpointStorage.sol";
+
+/**
+ * @title Burn module for burning tokens and keeping track of burnt amounts
+ */
+contract ScheduleCheckpoint is ScheduleCheckpointStorage, TransferManager, ICheckpoint {
+    using SafeMath for uint256;
+
+    event AddSchedule(bytes32 _name, uint256 _startTime, uint256 _endTime, uint256 _frequency, FrequencyUnit _frequencyUnit);
+    event RemoveSchedule(bytes32 _name);
+    event ModifyScheduleEndTime(bytes32 _name, uint256 _oldEndTime, uint256 _newEndTime);
+
+    /**
+     * @notice Constructor
+     * @param _securityToken Address of the security token
+     */
+    constructor(address _securityToken, address _polyToken) public Module(_securityToken, _polyToken) {
+
+    }
+
+    /**
+     * @notice This function returns the signature of configure function
+     */
+    function getInitFunction() public pure returns(bytes4) {
+        return bytes4(0);
+    }
+
+    /**
+     * @notice adds a new schedule for checkpoints
+     * @param _name name of the new schedule (must be unused)
+     * @param _startTime start time of the schedule (first checkpoint)
+     * @param _endTime End time of the schedule
+     * @param _frequency How frequent checkpoint will being created
+     * @param _frequencyUnit Unit of frequency i.e If issuer puts _frequency = 10
+     * & frequency unit is DAYS then it means every 10 day frequency new checkpoint will be created
+     */
+    function addSchedule(bytes32 _name, uint256 _startTime, uint256 _endTime, uint256 _frequency, FrequencyUnit _frequencyUnit) withPerm(OPERATOR) external {
+        require(_name != bytes32(0), "Empty name");
+        require(_startTime >= now, "Start time must be in the future");
+        require(schedules[_name].name == bytes32(0), "Name already in use");
+        uint256 endTime = _endTime;
+        if (_endTime <= _startTime)
```

Comment:
Shouldn't we require that either _endTime == 0, or _endTime > _startTime

Reply:
endTime is not a mandatory parameter, So I am not restricting an issuer about the endTime value. If they put anything that is below or equal to startTime it will make endTime = 0.